h/auth/util.py

```diff
@@ -117,8 +117,63 @@ def authority(request):
     return text_type(request.registry.settings.get('h.authority', request.domain))
 
 
+def check_auth_client(username, password, request):
```

Comment:
The actual logic within here is mostly a copy-and-paste job from the existing check.